I notice that $EPREFIX/usr/share/libtool/config/ltmain.sh also has this
stray increment - so I'd assume that this is a bug in libtool (which is
perhaps trying to fix a deprecated behaviour of old IRIX versions?).

http://bugs.gentoo.org/show_bug.cgi?id=301520


--- libltdl/config/ltmain.sh
+++ libltdl/config/ltmain.sh
@@ -6225,7 +6225,6 @@ func_mode_link ()
 	    current=$func_arith_result
 	    age="$number_minor"
 	    revision="$number_minor"
-	    lt_irix_increment=no
 	    ;;
 	  esac
 	  ;;
@@ -6297,11 +6296,7 @@ func_mode_link ()
 	  ;;
 
 	irix | nonstopux)
-	  if test "X$lt_irix_increment" = "Xno"; then
-	    func_arith $current - $age
-	  else
-	    func_arith $current - $age + 1
-	  fi
+	  func_arith $current - $age
 	  major=$func_arith_result
 
 	  case $version_type in
--- libltdl/config/ltmain.m4sh
+++ libltdl/config/ltmain.m4sh
@@ -6225,7 +6225,6 @@ func_mode_link ()
 	    current=$func_arith_result
 	    age="$number_minor"
 	    revision="$number_minor"
-	    lt_irix_increment=no
 	    ;;
 	  esac
 	  ;;
@@ -6297,11 +6296,7 @@ func_mode_link ()
 	  ;;
 
 	irix | nonstopux)
-	  if test "X$lt_irix_increment" = "Xno"; then
-	    func_arith $current - $age
-	  else
-	    func_arith $current - $age + 1
-	  fi
+	  func_arith $current - $age
 	  major=$func_arith_result
 
 	  case $version_type in
